okio/src/main/java/okio/Okio.java

```diff
@@ -242,4 +242,8 @@ static boolean isAndroidGetsocknameError(AssertionError e) {
     return e.getCause() != null && e.getMessage() != null
         && e.getMessage().contains("getsockname failed");
   }
+
+  public static Source tee(final Source source, final BufferedSink cacheBody) {
```

Comment:
You can use `Okio.buffer` to wrap a `Sink` with a `Buffer`. There doesn't seem to be a reason that one would need to be passed in.